x-pack/plugins/maps/public/classes/layers/raster_tile_layer/raster_tile_layer.ts

```diff
@@ -65,17 +61,7 @@ export class RasterTileLayer extends AbstractLayer {
     };
     const prevDataRequest = this.getSourceDataRequest();
     if (prevDataRequest) {
-      const prevMeta = prevDataRequest?.getMeta();
-      const canSkip = await canSkipSourceUpdate({
-        extentAware: false,
-        source,
-        prevDataRequest,
-        nextRequestMeta: nextMeta,
-        getUpdateDueToTimeslice: (timeslice?: Timeslice) => {
-          if (!prevMeta) return true;
-          return source.getUpdateDueToTimeslice(prevMeta, timeslice);
-        },
-      });
+      const canSkip = await source.canSkipSourceUpdate(prevDataRequest,nextMeta);
```

Comment:
Nick, I don't know the history behind the MVT_VECTOR layer, but looking at the code it appears that it was needed for the exact same reason that I have. The source has reasons that an update shouldn't be skipped. I would argue that an independent MVT_VECTOR layer wouldn't be needed if the source had the ability to let the layer know if it can skip an update, and specify when layers are stale. 

This same pattern is repeated in all the other vector layer types. [EMS_layer](https://github.com/elastic/kibana/blob/829c24d5b28315d67920a6cfad9374240f93c950/x-pack/plugins/maps/public/classes/layers/ems_vector_tile_layer/ems_vector_tile_layer.tsx#L119)

Pushing the canSkip/stale checks to the source, would allow us to not have to define a new layer type for every new source.

Reply:
Unless I'm missing some context, I believe that we could simplify this PR without modifying the parent classes. 

The url template is the only property in the `RasterTileSourceData` interface. So in the `syncData` method we can add a check to see if the url template has changed.

For example, if the url template is initially https://example.com/{z}/{x}/{y}.png?foo=foo. When the custom source changes the url template to https://example.com/{z}/{x}/{y}.png?foo=bar we can catch that and fetch the updated url. If there are no changes to the url, we skip the update.

I think it would look something like this.
```diff
--- a/x-pack/plugins/maps/public/classes/layers/raster_tile_layer/raster_tile_layer.ts
+++ b/x-pack/plugins/maps/public/classes/layers/raster_tile_layer/raster_tile_layer.ts
@@ -64,9 +64,12 @@ export class RasterTileLayer extends AbstractLayer {
       applyGlobalTime: source.getApplyGlobalTime(),
     };
     const prevDataRequest = this.getSourceDataRequest();
-    if (prevDataRequest) {
+    const prevData = (prevDataRequest?.getData() as RasterTileSourceData) ?? undefined;
+    if (prevData) {
+      const nextUrl = await source.getUrlTemplate(dataFilters);
+      const noChangesInSourceState: boolean = prevData.url === nextUrl;
       const prevMeta = prevDataRequest?.getMeta();
-      const canSkip = await canSkipSourceUpdate({
+      const noChangesInSearchState = await canSkipSourceUpdate({
         extentAware: false,
         source,
         prevDataRequest,
@@ -76,7 +79,7 @@ export class RasterTileLayer extends AbstractLayer {
           return source.getUpdateDueToTimeslice(prevMeta, timeslice);
         },
       });
-      if (canSkip) return;
+      if (noChangesInSourceState && noChangesInSearchState) return;
     }
     const requestToken = Symbol(`layer-source-refresh:${this.getId()} - source`);
     try {
```

Reply:
My specific source has params that are used for styles and params that are only used as metadata/metrics. The metadata/metrics shouldn't cause the source to be considered stale. 

meta data params are required for each query
meta_params = ['map_zoom', 'map_extent", "(and others)"]

this causes the url to look like
https://example.com/0/0/0.png?map_zoom=0.2&color_map=fire
https://example.com/0/0/0.png?map_zoom=0.4&color_map=fire
https://example.com/1/1/1.png?map_zoom=0.8&color_map=fire

the source is stale check causes a flicker because it does a comparison of 
'https://example.com/{z}/{x}/{y}.png?map_zoom=0.2&color_map=fire' === 'https://example.com/{z}/{x}/{y}.png?map_zoom=0.4&color_map=fire'

It is still the same exact tile until you hit the 0.5 zoom level
![image](https://user-images.githubusercontent.com/11495566/189214249-8264a773-6b73-4d7a-a6ec-306a3a4bd554.png)

So in my source the source isn't stale if the map_zoom changes, but is considered stale and should be removed/refreshed when color_map cahnged from fire to cool.

Another option is to allow custom layer types to be registered and we can then define any behavior we want for our sources. But that seemed much more complex since map_selector expects the layer type to be a known constant and there isn't currently a way to register custom layer types.

Reply:
Thank you for the additional context. This helps immensely. I'll discuss this with the team when we're all in the office next Tuesday.